src/DotNetty.Transport/Channels/AbstractChannel.cs

```diff
@@ -366,7 +366,7 @@ internal IMessageSizeEstimatorHandle EstimatorHandle
         /// <summary>
         /// {@link Unsafe} implementation which sub-classes must extend and use.
         /// </summary>
-        protected abstract class AbstractUnsafe : IChannelUnsafe
+        public abstract class AbstractUnsafe : IChannelUnsafe
```

Comment:
`protected` is enough unless `DefaultUnsafe` has to be public which is doesn't. E.g. it works fine for `AbstractSocketUnsafe` which also inherits from `AbstractUnsafe`.